design-patterns/creational-patterns/src/factory/Application.java

```diff
@@ -0,0 +1,22 @@
+package factory;
+
+public class Application {
+	static Archiver archiver;
+	
+	public static void initalize() {
+		String currentOs = System.getProperty("os.name");
+		
+		if (currentOs.equals("Windows")) {
+			archiver = new WindowsArchiver();
+		} else if (currentOs.equals("Linux")) {
+			archiver = new LinuxArchiver();
+		}
+	}
```

Comment:
I would suggest that this logic to be moved in separate class like : 
```java
ArchiverFactory.create(SystemType type)
```

because currently for me it looks like a Factory method pattern but not completed.